File: services/notification-service/src/main/java/com/pulsenotify/notification/model/Notification.java
package com.pulsenotify.notification.model;

import java.time.Instant;
import java.util.UUID;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.pulsenotify.events.NotificationChannel;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "notifications", schema = "notification")
public class Notification {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String recipient;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationChannel channel;

    private String subject;
    private String messageBody;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationStatus status;

    @CreationTimestamp
    private Instant createdAt;

    @UpdateTimestamp
    private Instant updatedAt;


}

File: frontend/src/types/notification.ts
export type NotificationChannel = 'EMAIL' | 'SMS' | 'PUSH'
export type NotificationStatus = 'PENDING' | 'SENT' | 'FAILED'

export interface Notification {
  id: string
  recipient: string
  subject: string | null
  channel: NotificationChannel
  messageBody: string
  status: NotificationStatus
  createdAt: string
  updatedAt: string
}

export interface CreateNotificationRequest {
  recipient: string
  subject?: string
  channel: NotificationChannel
  messageBody: string
}

export interface NotificationStats {
  total: number
  byStatus: Record<NotificationStatus, number>
  byChannel: Record<NotificationChannel, number>
}
